pkg/controller/dataimportcron-controller.go

```diff
@@ -311,13 +314,40 @@ func (r *DataImportCronReconciler) update(ctx context.Context, dataImportCron *c
 	dataImportCronCopy := dataImportCron.DeepCopy()
 	imports := dataImportCron.Status.CurrentImports
 	importSucceeded := false
+
+	dataVolume := dataImportCron.Spec.Template
+	explicitScName := getStorageClassFromTemplate(&dataVolume)
+	dvStorageClass, err := cc.GetStorageClassByName(r.client, explicitScName)
+	if err != nil {
+		return res, err
+	}
+	format, err := r.getSourceFormat(ctx, dataImportCron, dvStorageClass)
+	if err != nil {
+		return res, err
+	}
+	snapshot, err := r.getSnapshot(ctx, dataImportCron, format)
+	if err != nil {
+		return res, err
+	}
+
+	handleDataVolumePopulated := func() error {
+		if err = r.updateSourceObject(ctx, dataImportCron, pvc); err != nil {
+			return err
+		}
+		importSucceeded = true
+		if err := r.handleCronFormat(ctx, dataImportCron, format, dvStorageClass); err != nil {
```

Comment:
why do we need this helper (used only here)? if Snapshot just call r.handleSnapshotFormat() directly

Reply:
just trying to be more welcoming towards new formats, I thought it's nice to have a structured entrypoint

Reply:
so maybe move the handleDataVolumePopulated logic to handleCronFormat and name it accordingly

Reply:
handlePopulatedPvc (coming soon) doesn't belong there imo, why is this so bad?